### PR TITLE
[Refactor] Simplify EngineTask

### PR DESCRIPTION
**Pull request by decster**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

EngineTask's methods like `prepare/finish/cancel/get_related_tablets` do not abstract anything useful(not used anywhere, or it was useful but was removed later), so this PR removes those.

### CODE REVIEW

**Review by sduzh — DISMISSED** [review]

**Comment by sduzh** [comment]
LGTM, +1

**Review by Linkerist — DISMISSED** [review]

**Review by trueeyu — DISMISSED** [review]

File: be/src/storage/task/engine_publish_version_task.cpp [hunk 1]
```diff
@@ -37,7 +37,7 @@ EnginePublishVersionTask::EnginePublishVersionTask(TTransactionId transaction_id
           _tablet_info(tablet_info),
           _rowset(rowset) {}
 
-Status EnginePublishVersionTask::finish() {
```
Comment (imay) [inline be/src/storage/task/engine_publish_version_task.cpp:40]: before this change, finish is executed with all tablet lock held. after this change, lock is not held. Are you sure is it OK to do this change?

Comment (decster) [reply be/src/storage/task/engine_publish_version_task.cpp:40]: `before this change, finish is executed with all tablet lock held`
No, which part of the code are you talking about?

**Review by chaoyli — DISMISSED** [review]

File: be/src/storage/task/engine_publish_version_task.cpp (hunk 1 above)
Comment (decster) [reply be/src/storage/task/engine_publish_version_task.cpp]: get_related_tablets returns empty, so it doesn't change anything

**Review by imay — DISMISSED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**